modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -596,9 +597,14 @@ public CommitData createCommitData() {
 
   @Override
   public synchronized void commit() throws CommitException {
-    SyncCommitObserver sco = new SyncCommitObserver();
-    commitAsync(sco);
-    sco.waitForCommit();
+    SyncCommitObserver sco = null;
+    try {
+      sco = new SyncCommitObserver();
+      commitAsync(sco);
+      sco.waitForCommit();
+    } finally {
+      sco = null;
```

Comment:
Oh, I didn't notice they were final.  Interesting, clear has a cost it loops internally setting stuff to null.  I suppose this is ok?  Maybe its more little more work here and then less work later for the GC, not sure.

I thought about removing final, but final is nice for ensuring correctness.  So I am not inclined to do that.